### DISCUSSION TITLE
Регулирование настроек Meilisearch

### DISCUSSION THREAD

**Discussion by smplseven** [totumonline/totum-ru-issues-and-discussions#353] [Есть идея, а что если...]
Добрый день! Мы сейчас используем в нашем внутреннем проекте автоматизации общий поиск по системе (Meilisearch), кстати, спасибо, он очень удобно сделан, но обнаружили некоторую "фичу", которая в нашем кейсе использования является скорее проблемой в безопасности. 
Системой пользуются несколько ролей, для которых доступны разные таблицы и разные записи внутри таблиц. При этом на всех таблицах висит общий поиск.
Также в всплывающее окно поиска мы сразу добавили некоторую информацию по найденной записи. 

Изначально поиск доступен всем сотрудникам и, несмотря на то, что сотруднику закрыт какой-то раздел или запись, то путем перебора он может в общем поиске проверить, есть ли значение в нашей таблице или нет и это для нас проблема в безопасности. 

Возможно ли дополнить настройки Meilisearch ролевой моделью, которая позволит выдавать в поиске значения, если только пользователь имеет право видеть это значение в таблице?

**Comment by ttm-alexeypolunin** [comment]
А вы прям проэксперементировали и получили те результаты, которые пользователи не должен видеть?

Тк там есть автоматические ограничения накладывающиеся на хиты перед отправкой на клиент. Проверяется:

— доступ к таблице
— доступ к конкретной строке

Под конкретной строкой понимается ограничение доступа через Префильтр — если префильтр есть и он заблокировал для пользователя на изменения то подобранные результаты поиска проверяются через эти префильтры перед выдачей.

Фильтрации подвергается вся плашка хита.

**Reply by ttm-alexeypolunin** [reply]
Возможно про что мы забыли — это ролевой доступ к циклам для режима Видит только свои циклы.

Те серч не может работать внутри цикла, только на уровне таблицы Циклов, но возможно мы забыли чекнуть это место — посмотрим.

**Reply by smplseven** [reply]
Большое спасибо за развернутый ответ !
Видимо  мы упустили этот момент в документации - теперь всё проверили и всё ок!

**Comment by poluninatm** [comment]
Проверила: проверка доступа к циклам тоже подключена
